### ISSUE TITLE
when items is bound to a property that is modified with ele.push it sorta breaks.

### ISSUE THREAD

**Issue by ktiedt** [vaadin/vaadin-combo-box#127]
I had a custom element that uses a combo-box internally and the items array was generated by ele.push as the parent element was initializing... when I did this, the combo-box would only ever show the first value as its only option, once I selected the first value though, I could then see all of the options (and debugging the items value, it definitely contained all of the items from the start of this test).

**Comment by Saulis** [comment]
Hi!

Could you be kind enough to provide a JSBin or similar that we could use to make sure we get the exact issue you're facing fixed? 

I believe you're talking about calling `combobox.push('items', 'foo');` and at least when the overlay is opened, it doesn't update correctly, right? (I'm using `1.0.0-alpha4`)

As a workaround, you should be able to call `combobox.$.overlay.push('_items', 'foo');` to make the list update correctly. We'll add the missing observers to `combobox.items` asap.

**Comment by ktiedt** [comment]
I'll try and get a JSBin going based on my example at work tomorrow, however what you stated is not quite what we were doing.

We had something more like: (heavily paraphrased)
```
<dom-module>
  <vaadin-combo-box items="[[searchItems]]">
</dom-module>
```

and then were doing this.push("searchItems", .....); in our custom element.

**Comment by Saulis** [comment]
OK, thanks for the clarification!

In any case, I believe we need to get `combobox.push('items'...)` to work before the data binding you described can work.

As a workaround then, you'd probably need to add an observer for `searchItems.*` which always replaces the whole `items` by doing `this.$.combobox.items = this.searchItems`

**Comment by ktiedt** [comment]
We actually swapped `searchItems` to a computed, due to a different problem and that solved it, but it seemed like something worth reporting and fixing :)

**Comment by Saulis** [comment]
OK interesting.. might be that with a computed value the whole array is always replaced when it's recomputed.

**Comment by ktiedt** [comment]
That is my thought too.

**Comment by Saulis** [comment]
Fixed in #133
